### PR TITLE
Router transformer end point for marketo

### PR DESCRIPTION
**Pull request by ruchiramoitra**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by sayan-mitra** [comment]
let's pass the version in routerHandleDest as a parameter

File: v0/destinations/marketo/transform.js [hunk 1]
```diff
@@ -293,16 +288,110 @@ const processEvent = async (message, destination) => {
...
+  returnResponse.message = message;
+  returnResponse.metadata = metadata;
+  returnResponse.batched = false;
+  returnResponse.statusCode = 200;
+  returnResponse.destination = destination;
+  return returnResponse;
+};
+
+// Error responses
+const getErrorRespEvents = (metadata, statusCode, error) => {
```
Comment (sayan-mitra) [inline v0/destinations/marketo/transform.js:320]: this is returning an object, can we have a better name?

File: v0/destinations/marketo/transform.js [hunk 2]
```diff
@@ -293,16 +288,110 @@ const processEvent = async (message, destination) => {
...
+    return [respEvents];
+  }
+  let token;
+  try {
+    token = await getAuthToken(formatConfig(input[0].destination));
+  } catch (error) {
+    const respEvents = getErrorRespEvents(
+      input.map(ev => ev.metadata),
+      error.response ? error.response.status : 400,
+      error.message || "Error occurred while processing payload."
```
Comment (sayan-mitra) [inline v0/destinations/marketo/transform.js:347]: Error while fetching token

File: v0/destinations/marketo/transform.js [hunk 3]
```diff
@@ -293,16 +288,110 @@ const processEvent = async (message, destination) => {
...
+    );
+    return [respEvents];
+  }
+  // If token is null track/identify calls cannot be executed.
+  if (!token) {
```
Comment (sayan-mitra) [inline v0/destinations/marketo/transform.js:352]: do we need this separately
can this be moved inside above catch block

File: v0/destinations/marketo/transform.js [hunk 4]
```diff
@@ -293,16 +288,110 @@ const processEvent = async (message, destination) => {
...
+  // Checking previous status Code. Initially setting to false.
+  // If true then previous status is 500 and every subsequent event output should be sent with status code 500 to the router to be retried.
+  let prevStatus = false;
+  for (let i = 0; i < input.length; i += 1) {
+    const inputs = input[i];
+    let respEvents = {};
+    if (prevStatus) {
+      respEvents = getErrorRespEvents(
+        [inputs.metadata],
+        500,
```
Comment (sayan-mitra) [inline v0/destinations/marketo/transform.js:370]: this case will never hit right?
since we are making an external call at the very first, do we need this?

Comment (sayan-mitra) [reply v0/destinations/marketo/transform.js:370]: if we need, can we do a continue instead of if-else

**Review by arnab-p — APPROVED** [review]

**Pull request merged**